Server/mods/deathmatch/logic/packets/CPlayerPuresyncPacket.cpp

```diff
@@ -175,6 +175,12 @@ bool CPlayerPuresyncPacket::Read(NetBitStreamInterface& BitStream)
             // Set weapon slot
             if (bWeaponCorrect)
                 pSourcePlayer->SetWeaponSlot(uiSlot);
+            else
+            {
+                // remove invalid weapon data to prevent this from being relayed to other players
```

Comment:
I think it would be nice to add a comment explaining what could cause this (Cheaters, etc).